### PR TITLE
Validation Client rewrite

### PR DESCRIPTION
**Pull request by jonfung-dydx**
With the addition of the new `fns-snapshot-interval` flag in the protocol, we can now vastly simplify the validation logic code.

### CODE REVIEW

File: src/feed_handler.py [hunk 1]
```diff
@@ -261,3 +265,25 @@ def get_books(self) -> Dict[int, lob.LimitOrderBook]:
...
+            logging.error(f"Mismatched book clob pair ids: \n{self_books.keys()}\n{other_books.keys()}")
+            return False
+
+        for cpid in self_books.keys():
+            self_book = self_books[cpid]
+            other_book = other_books[cpid]
+            comparison_failed = self_book.compare_books(other_book)
+            if comparison_failed:
+                failed = True
+        # TODO compare subaccounts
```
Comment (dydxwill) [inline src/feed_handler.py:288]: remove todo?

File: src/helpers.py [hunk 2]
```diff
@@ -65,3 +66,19 @@ def parse_protocol_order(order: Order) -> lob.Order:
...
+    """
+    Checks if the given update is a snapshot update.
+    """
+    update_type = update.WhichOneof('update_message')
+    if update_type == 'orderbook_update':
+        return update.orderbook_update.snapshot
+    # Fills are never emitted 
+    elif update_type == 'order_fill':
+        return False
+    elif update_type == 'taker_order':
```
Comment (dydxwill) [inline src/helpers.py:79]: combine if conditions

**Review by dydxwill — APPROVED** [review]

**Comment by yang-dydx** [comment]
Why do we want to remove all the non-snapshot validation logic? If the non-snapshot based logic breaks will we still be able to catch it?

**Review by jayy04 — APPROVED** [review]

**Comment by matthewdowney** [comment]
> Why do we want to remove all the non-snapshot validation logic? If the non-snapshot based logic breaks will we still be able to catch it?

Spoke with @jonfung-dydx offline about this, he says this method has equal coverage to the previous one, so sounds good to me.

**Review by matthewdowney — APPROVED** [review]

File: src/feed_handler.py (hunk 1 above)
Comment (jonfung-dydx) [reply src/feed_handler.py:325]: Think it has to be done later, will keep after you add this functionality

**Pull request merged**